test/behaviour/connection/ConnectionStepsBase.java

```diff
@@ -72,6 +84,8 @@ void before() {
         assertNotNull(address);
         client = createTypeDBClient(address);
         client.databases().all().forEach(Database::delete);
+        sessionOptions = createOptions().infer(true);
+        transactionOptions = createOptions().infer(true);
```

Comment:
we always turn inference on - this was extracted from lower level implementation where we opened the transaction